docs/src/submodules/Bridges/implementation.md

```diff
@@ -45,3 +45,24 @@ get(::Bridges.Constraint.AbstractBridge, ::ListOfVariableIndices)
 get(::Bridges.AbstractBridge, ::NumberOfConstraints)
 get(::Bridges.AbstractBridge, ::ListOfConstraintIndices)
 ```
+
+# SetMap bridges
+
+Implementing a constraint bridge relying on linear transformation between two
+sets is easier thanks to the [SetMap interface](constraint_set_map).
+The bridge simply needs to be a subtype of [`Bridges.Constraint.SetMap`] and
+the linear transformation is represented with [`Bridges.Constraint.map_set`](@ref),
+[`Bridges.Constraint.map_function`](@ref),
+[`Bridges.Constraint.inverse_map_set`](@ref),
+[`Bridges.Constraint.inverse_map_function`](@ref),
+[`Bridges.Constraint.adjoint_map_function`](@ref) and
+[`Bridges.Constraint.inverse_adjoint_map_function`](@ref).
+Note that the implementing last 4 methods is optional in the sense that if they
+are not implemented, bridging constraint would still work but some features
+would be missing as described in the docstrings.
+See [L20, Section 2.1.2] for more details including [L20, Example 2.1.1] that
+illustrates the idea for
```

Comment:
Oh hey. This actually made it into your thesis? I should read.

Reply:
Yes, the section 2.1.2 is meant to be the howto guide for how to implement the ConstraintDual and ConstraintDualStart. It's doable for SetMap bridges but for more complex cases like QuadtoSOC or when you create auxiliary variables it can get complicated (see e.g. the comments in https://github.com/jump-dev/MathOptInterface.jl/pull/1178, https://github.com/jump-dev/MathOptInterface.jl/pull/1017, https://github.com/jump-dev/MathOptInterface.jl/pull/1021). The eq. (2.8) of the thesis gives an automatic way to get this which is much easier than trying to do ad-hoc arguments for each bridge.